packit_service/worker/github_handlers.py

```diff
@@ -517,3 +460,130 @@ def run(self) -> HandlerResults:
                 )
 
         return HandlerResults(success=True, details={})
+
+
+@add_to_pr_comment_mapping
+class GitHubPullRequestCommentCoprBuildHandler(PullRequestCommentHandler):
+    """ Issue handler for comment `/packit copr-build` """
+
+    name = PullRequestCommentAction.copr_build
+
+    def __init__(self, config: Config, event: PullRequestCommentEvent):
+        super(GitHubPullRequestCommentCoprBuildHandler, self).__init__(
+            config=config, event=event
+        )
+        self.config = config
+        self.event = event
+        self.project: GitProject = self.github_service.get_project(
+            repo=event.base_repo_name, namespace=event.base_repo_namespace
+        )
+        # Get the latest pull request commit
+        self.event.commit_sha = self.project.get_all_pr_commits(self.event.pr_id)[-1]
+        self.package_config: PackageConfig = get_package_config_from_repo(
+            self.project, self.event.commit_sha
+        )
+        self.package_config.upstream_project_url = event.https_url
+
+    def get_tests_for_build(self) -> Optional[JobConfig]:
+        """
+        Check if there are tests defined
+        :return: JobConfig or None
+        """
+        for job in self.package_config.jobs:
+            if job.job == JobType.tests:
+                return job
+        return None
+
+    def run(self) -> HandlerResults:
+
+        collaborators = self.project.who_can_merge_pr()
+        r = BuildStatusReporter(self.project, self.event.commit_sha)
+        if self.event.github_login not in collaborators:
+            msg = "Only collaborators can trigger Packit-as-a-Service"
+            r.set_status("failure", msg, PRCheckName.get_build_check())
+            return HandlerResults(success=False, details={"msg": msg})
+
+        cbh = CoprBuildHandler(
+            self.config, self.package_config, self.project, self.event
+        )
+        handler_results = cbh.run_copr_build()
+        if handler_results["success"]:
+            # Testing farm is triggered just once copr build is finished as it uses copr builds
+            # todo: utilize fedmsg for this.
+            test_job_config = self.get_tests_for_build()
+            if test_job_config:
+                testing_farm_handler = GithubTestingFarmHandler(
+                    self.config, test_job_config, self.event
+                )
+                testing_farm_handler.run()
+            else:
+                logger.debug("Testing farm not in the job config.")
+            return HandlerResults(success=True, details={})
+        return handler_results
+
+
+@add_to_pr_comment_mapping
+class GitHubPullRequestCommentBuildHandler(PullRequestCommentHandler):
```

Comment:
Now that I look at this, I probably don't follow the E2E workflow: should people type `/packit build` to get a prod koji build of HEAD of dist-git master? My concern is that the PR is completely irrelevant since it's not used during the build in any way.

Reply:
The code is commented out.

Reply:
I don't follow the conversation/PR to a detail (due to it's being too huge), but commenting out such a big parts of code usually isn't a good solution.

Reply:
Yesterday, we decided to comment out on grooming.